app/src/main/java/com/github/polypoly/app/base/PastGame.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.polypoly.app.base
+
+import com.github.polypoly.app.base.user.User
+
+/**
+ * A class that represent a past [Game] played by [User]s
+ * @property users The id of the [User]s that played the game
+ * @property usersRank The rank of the [User]s in the [Game] by id
+ * @property date The date when the [Game] began in Unix-based time
+ * @property duration The duration of the [Game] in seconds
+ */
+data class PastGame(
+    private val users: List<User> = listOf(),
+    private val usersRank: Map<Long, Int> = mapOf(),
+    val date: Long = 0,
+    val duration: Long = 0,
+) {
+
+    /**
+     * Return if the [User] has won the [Game]
+     * @param userId the id of the [User]
+     * @return true if the user has won the [Game], false otherwise
+     */
+    fun hasWin(userId: Long): Boolean {
+        return usersRank[userId] == 1
+    }
+
+    /**
+     * get the [User] id of the winner
+     * @return the [User] id of the winner
+     */
+    fun getWinnerId(): Long {
+        return usersRank.filterValues { it == 1 }.keys.first()
```

Comment:
Here is another example of easy Magic Values to avoid, you could create a `WINNER = 1` constant :)!

Reply:
In fact the 1 does not represent the winner but the rank of the player. And when you are first, you are the winner. I don't know what you think about it but here I think the 1 makes sense.